From 235fc0e36d35270e1ff426f26bbea6154b8863a5 Mon Sep 17 00:00:00 2001
Message-Id: <235fc0e36d35270e1ff426f26bbea6154b8863a5.1648734704.git.geliang.tang@suse.com>
From: Colin Ian King <colin.king@canonical.com>
Date: Fri, 26 Mar 2021 19:43:48 +0000
Subject: [PATCH] bpf: Remove redundant assignment of variable id
Git-commit: 235fc0e36d35270e1ff426f26bbea6154b8863a5
Patch-mainline: v5.13-rc1
References: bsn#127
Modified-by-SEL: No


The variable id is being assigned a value that is never read, the
assignment is redundant and can be removed.

Addresses-Coverity: ("Unused value")
Signed-off-by: Colin Ian King <colin.king@canonical.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Acked-by: Song Liu <songliubraving@fb.com>
Link: https://lore.kernel.org/bpf/20210326194348.623782-1-colin.king@canonical.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/bpf/btf.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/kernel/bpf/btf.c b/kernel/bpf/btf.c
index ec8afc4bc560..0600ed325fa0 100644
--- a/kernel/bpf/btf.c
+++ b/kernel/bpf/btf.c
@@ -789,7 +789,6 @@ static const struct btf_type *btf_type_skip_qualifiers(const struct btf *btf,
 
 	while (btf_type_is_modifier(t) &&
 	       BTF_INFO_KIND(t->info) != BTF_KIND_TYPEDEF) {
-		id = t->type;
 		t = btf_type_by_id(btf, t->type);
 	}
 
-- 
2.34.1

